src/test_l1_dsHdmiIn.c

```diff
@@ -2016,7 +2029,13 @@ void test_l1_dsHdmiIn_positive_dsGetSupportedGameFeaturesList(void) {
     UT_ASSERT_EQUAL(dsGetSupportedGameFeaturesList(&supported_features_1), dsERR_NONE);
 
     // Step 3: Call dsGetSupportedGameFeaturesList() with valid inputs (dsSupportedGameFeatureList_t*)
-    UT_ASSERT_EQUAL(dsGetSupportedGameFeaturesList(&supported_features_2), dsERR_NONE);
+    ut_kvp_getStringField(instance, "dsHdmiIn/gameFeatures/feature", featureString, sizeof(featureString));
+
+    strncpy(supported_features_2.gameFeatureList, featureString, sizeof(supported_features_2.gameFeatureList) - 1);
```

Comment:
You need a #define for the sizeof supported_features_2.gameFeatureList, and therefore everywhere when you use `strncpy( xxx, xxx, DS_HDMI_IN_GAME_FEATURE_LIST_SIZE )...`

or ideally define a common `DS_HDMI_IN_MAX_ARRAY_SIZE`

Reply:
Instead of this one, I have used the macro
UT_ASSERT_KVP_EQUAL_PROFILE_STRING
UT_ASSERT_KVP_EQUAL_PROFILE_UINT32